auslib/db.py

```diff
@@ -928,16 +929,20 @@ def __init__(self, db, dialect, metadata, baseTable, conditions=("time", "uptake
             # See https://github.com/zzzeek/sqlalchemy/blob/rel_0_7/lib/sqlalchemy/schema.py#L781
             # for background.
             newcol.key = newcol.name = "base_%s" % col.name
-            # 2) Primary keys on the base table are normal columns here,
-            # and are allowed to be null (because we support adding new rows
-            # to tables with autoincrementing keys via scheduled changes).
-            # The base table's data version may also be null for the same reason.
-            if col.primary_key or newcol.name == "base_data_version":
+            # 2) All non PK columns from baseTable should be nullable. Base table PK columns
+            # which are autoincrement should be nullable in the Scheduled Changes version of
+            # the column. Base table PK columns which are not autoincrement should not be nullable.
```

Comment:
Sorry - missed this comment until just now. Just a couple of small clarifications:
> For Primary Key columns that aren't Integer or Autoincrement but are nullable

When you say "but are nullable" are you talking about the base table or the scheduled changes table? It wouldn't be possible to make it nullable in the base table, because it woul be part of the key.

> Non-Primary Key columns from the baseTable become nullable and non-unique because a
> scheduled deletion will have null values in all non-PK even if they are non-nullable or
> unique in the baseTable.

100% right on the nullable part. When it comes to uniqueness, it's a bit more subtle. Take a case like like rules.alias. In the Rules table, we allow alias to be used in place of rule_id in some places, so uniqueness is necessary. A field like alias is used in the business logic of Rules, and that business logic requires things like uniqueness to be in place. In rules_scheduled_changes, things are different. The base table columns are _not_ really part of the business logic of a Scheduled Changes table. We need to store values in them, but the real job of Scheduled Changes is putting in place new versions of a base table row upon certain conditions being met. To do that job, Scheduled Changes needs to keep track of conditions and signoffs, but the base table columns are more or less just simple data storage.

That's probably more info than you need for the comment update, but hopefully it's helpful background.